pulp_rpm/app/serializers.py

```diff
@@ -322,3 +325,90 @@ class OneShotUploadSerializer(serializers.Serializer):
         help_text=_("The rpm file."),
         required=True,
     )
+
+
+class CopySerializer(serializers.Serializer):
+    """
+    A serializer for Content Copy API.
+    """
+
+    source_repo = serializers.HyperlinkedRelatedField(
+        help_text=_('A URI of the repository.'),
+        required=False,
+        queryset=Repository.objects.all(),
+        view_name='repositories-detail',
+    )
+    source_repo_version = NestedRelatedField(
+        help_text=_('A URI of the repository version'),
+        required=False,
+        queryset=RepositoryVersion.objects.all(),
+        parent_lookup_kwargs={'repository_pk': 'repository__pk'},
+        lookup_field='number',
+        view_name='versions-detail',
+    )
+    dest_repo = serializers.HyperlinkedRelatedField(
+        help_text=_('A URI of the repository.'),
+        required=True,
+        queryset=Repository.objects.all(),
+        view_name='repositories-detail',
+    )
+    types = serializers.ListField(
+        help_text=_('A list of types to copy ["package", "errata"]'),
+        write_only=True,
+        default=['package', 'errata']
+    )
+
+    def validate(self, data):
+        """
+        Validate that the Serializer contains valid data.
+
+        Set the Repository based on the RepositoryVersion if only the latter is provided.
+        Set the RepositoryVersion based on the Repository if only the latter is provied.
+        Convert the human-friendly names of the content types into what Pulp needs to query on.
+        """
+
+        super().validate(data)
+        if hasattr(self, 'initial_data'):
+            validate_unknown_fields(self.initial_data, self.fields)
+
+        new_data = {}
+        new_data.update(data)
+
+        source_repo = data.get('source_repo')
+        source_repo_version = data.get('source_repo_version')
+
+        if not source_repo and not source_repo_version:
```

Comment:
An alternative way as an attempt to improve readability (not critical and maybe not a better option):
How about to eliminate bad combinarions first and then deal with the rest one by one?
```
if repo and repo_version:
    raise ...

if not repo and not repo_version:
    raise ...

if repo:
    ...

if repo_version:
    ...